packages/cardpay-sdk/sdk/safes/base.ts

```diff
@@ -267,24 +281,40 @@ export default class Safes {
       estimate.gasToken,
       ZERO_ADDRESS
     );
+
+    let txnHash = result.ethereumTx.txHash;
+
     if (typeof onTxnHash === 'function') {
-      await onTxnHash(result.ethereumTx.txHash);
+      await onTxnHash(txnHash);
     }
-    return await waitUntilTransactionMined(this.layer2Web3, result.ethereumTx.txHash);
+    return await waitUntilTransactionMined(this.layer2Web3, txnHash);
   }
 
+  async setSupplierInfoDID(txnHash: string): Promise<TransactionReceipt>;
   async setSupplierInfoDID(
     safeAddress: string,
     infoDID: string,
     gasToken: string,
     txnOptions?: TransactionOptions,
     contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async setSupplierInfoDID(
+    safeAddressOrTxnHash: string,
+    infoDID?: string,
+    gasToken?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
   ): Promise<TransactionReceipt> {
+    if (!infoDID) {
+      let txnHash = safeAddressOrTxnHash;
+      return waitUntilTransactionMined(this.layer2Web3, txnHash);
+    }
+    let safeAddress = safeAddressOrTxnHash;
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
     let from = contractOptions?.from ?? (await this.layer2Web3.eth.getAccounts())[0];
     let supplierManager = await getAddress('supplierManager', this.layer2Web3);
-    let payload = await this.setSupplierInfoDIDPayload(infoDID);
-    let estimate = await gasEstimate(this.layer2Web3, safeAddress, supplierManager, '0', payload, 0, gasToken);
-    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    let payload = await this.setSupplierInfoDIDPayload(infoDID!);
```

Comment:
ditto with the comments above around using a test to assert this exists instead of using !'s